### DISCUSSION TITLE
Limit 1 when using search plugin?

### DISCUSSION THREAD

**Discussion by aronymous** [Flexget/Flexget#3744] [Q&A]
I have recently started using the IPtorrents search plugin to help clean up my movie list for items that is not on my preferred tracker. Everything works except if it finds multiple results that match it tries to download all of them. 
Here is the log of it accepting multiple entries for the same movie. 

2023-04-17 21:42:35 VERBOSE  task          iptold          ACCEPTED: `Annabelle Creation 2017 BluRay 1080p DTS x264-PRoDJi` by regexp plugin because regexp '2023' didn't match
2023-04-17 21:42:35 VERBOSE  task          iptold          ACCEPTED: `Annabelle Creation 2017 1080p BluRay REMUX AVC Atmos-EPSiLON` by regexp plugin because regexp '2023' didn't match
2023-04-17 21:42:35 VERBOSE  task          iptold          ACCEPTED: `Annabelle Creation 2017 1080p BluRay x264-WiKi` by regexp plugin because regexp '2023' didn't match

Here is my config for the task

    trakt_lookup: yes
    discover:
      what:
        - movie_list: trackt
      from:
        - iptorrents:
            rss_key: xxxxx
            uid: xxxx
            password: xxxx
            category:
            - Movie-BD-Rip
            - Movie-HD-Bluray
    quality: 1080p bluray
    regexp:
      reject:
        - 2023:
            from: movie_year
        - 2022:
            from: movie_year
    torrent_alive: 10
    list_match:
      from:
        - movie_list: movies
    template:
      - rtor_add
      - slack_not
    list_remove:
      - trakt_list:
          username: xxxx
          account: xxxx
          list: Movies

Is there a way to tell this task to only accept 1 result per search?
I thought of using the limit plugin but not sure if its what I need.

**Comment by BrutuZ** [comment]
`limit` would make `discover` stop after a single entry. If that entry happened to not pass your filters it wouldn't have other entries to try.
You might want to look into [`limit_new`](https://flexget.com/Plugins/limit_new) instead.

Bonus round: You can have the year rejection in a single expression with `202[23]` covering both years or even shorten and future-proof it further by using `if` instead of `regexp`
```yaml
if:
  - 'movie_year|d(now.year) >= 2022': reject
```

**Comment by gazpachoking** [comment]
I believe list_match has some options to only allow one match per item
